app/forms/incomeSources/add/BusinessStartDateForm.scala

```diff
@@ -0,0 +1,106 @@
+  private val dateMustBeReal = s"$messagePrefix.error.invalid"
+  private val dayRequired = s"dateForm.error.day.required"
+  private val monthRequired = s"dateForm.error.month.required"
+  private val yearRequired = s"dateForm.error.year.required"
+  private val dayAndMonthRequired = s"dateForm.error.dayAndMonth.required"
+  private val dayAndYearRequired = s"dateForm.error.dayAndYear.required"
+  private val monthAndYearRequired = s"dateForm.error.monthAndYear.required"
+
+  val day: String = "day"
+  val month: String = "month"
+  val year: String = "year"
+
+  val form: Form[DateFormElement] = {
+    val currentDate: LocalDate = dateService.getCurrentDate()
+    val currentDatePlusOneWeekOneDay: LocalDate = currentDate.plusWeeks(1).plusDays(1)
+    val futureErrorMessage: String = dateFormatter.longDate(currentDatePlusOneWeekOneDay)(messages).toLongDate
+    def dateMustNotBeInTheFuture(maximumDate: String): String = messages(dateMustNotBeTooFarInFuture, maximumDate)
+
+    Form(
+      mapping(s"$messagePrefix" -> tuple(
+        day -> default(text(), ""),
+        month -> default(text(), ""),
+        year -> default(text(), ""))
+        .verifying(firstError(
+          checkRequiredFields,
+          validDate(dateMustBeReal)
+        )).transform[LocalDate](
+        {
+          case (day, month, year) => LocalDate.of(year.toInt, month.toInt, day.toInt)
+        },
+        date => (date.getDayOfMonth.toString, date.getMonthValue.toString, date.getYear.toString)
+      )
+        .verifying(maxDate(currentDatePlusOneWeekOneDay, dateMustNotBeInTheFuture(futureErrorMessage)))
+      )(DateFormElement.apply)(DateFormElement.unapply))
+  }
+
+  private def checkRequiredFields: Constraint[(String, String, String)] = Constraint("constraints.requiredFields") {
+    case (day, month, year) if day.trim.isEmpty && month.trim.isEmpty && year.trim.isEmpty =>
+      Invalid(Seq(ValidationError(dateMustBeEntered)))
+    case (day, month, year) if day.trim.isEmpty && month.trim.nonEmpty && year.trim.isEmpty =>
+      Invalid(Seq(ValidationError(dayAndYearRequired)))
+    case (day, month, year) if day.trim.isEmpty && month.trim.isEmpty && year.trim.nonEmpty =>
+      Invalid(Seq(ValidationError(dayAndMonthRequired)))
+    case (day, month, year) if day.trim.nonEmpty && month.trim.isEmpty && year.trim.isEmpty =>
+      Invalid(Seq(ValidationError(monthAndYearRequired)))
+    case (day, month, year) if day.trim.isEmpty && month.trim.nonEmpty && year.trim.nonEmpty =>
+      Invalid(Seq(ValidationError(dayRequired)))
+    case (day, month, year) if day.trim.nonEmpty && month.trim.isEmpty && year.trim.nonEmpty =>
+      Invalid(Seq(ValidationError(monthRequired)))
+    case (day, month, year) if day.trim.nonEmpty && month.trim.nonEmpty && year.trim.isEmpty =>
+      Invalid(Seq(ValidationError(yearRequired)))
+    case _ =>
+      Valid
+  }
+
+  override protected def maxDate(maximum: LocalDate, errorKey: String, args: Any*): Constraint[LocalDate] =
```

Comment:
Not sure why this has been overridden?

Reply:
The requirements are to stop entry above 7 days, so it should be after (not equals) as per my implementation of this function